components/Pages/Project/AirdropGitcoinSupporters.tsx

```diff
@@ -0,0 +1,446 @@
+    const [metadata, setMetadata] = useState<any>(null)
+    const [customDescription, setCustomDescription] = useState("");
+    const { writeContract, data: txData, isPending: isMinting, error: mintError, isSuccess, } = useWriteContract()
+    const { switchChainAsync } = useSwitchChain();
+
+    const chainId = useChainId()
+
+    const handleImageUpload = async (event: React.ChangeEvent<HTMLInputElement>) => {
+        if (event.target.files && event.target.files[0]) {
+            setFileUploading(true);
+            const formData = new FormData();
+            formData.append('file', event.target.files[0]);
+
+            await axios.post('https://api.pinata.cloud/pinning/pinFileToIPFS', formData, {
+                headers: {
+                    'Authorization': `Bearer ${envVars.IPFS_TOKEN}`,
+                    'Content-Type': 'multipart/form-data'
+                }
+            }).then(async (response) => {
+                setImageIPFSHash(response.data.IpfsHash);
+                console.log('File uploaded to IPFS with hash:', response.data.IpfsHash);
+
+                const defaultDescription = `This NFT is issued to honor and recognize your invaluable support of donors who have generously funded ${projectDetails.project.metadata.title}, symbolizing their crucial role in driving impactful change.`;
+                const fullDescription = customDescription ? `${customDescription} \n|\n ${defaultDescription}` : defaultDescription;
+
+                const metadata = {
+                    name: `${projectDetails.project.metadata.title} - Karma GAP Patron`,
+                    description: fullDescription,
+                    image: `ipfs://${response.data.IpfsHash}`,
+                    attributes: []
+                };
+
+                await axios.post('https://api.pinata.cloud/pinning/pinJSONToIPFS', {
+                    pinataContent: metadata,
+                    pinataMetadata: {
+                        name: "Karma GAP Patron NFT Metadata"
+                    }
+                }, {
+                    headers: {
+                        'Authorization': `Bearer ${envVars.IPFS_TOKEN}`,
+                        'Content-Type': 'application/json'
+                    }
+                }).then((metadataResponse) => {
+                    setMetadata(metadata)
+                    setMetadataIPFSHash(metadataResponse.data.IpfsHash);
+                    console.log('Metadata uploaded to IPFS with hash:', metadataResponse.data.IpfsHash);
+                    setFileUploading(false);
+                }).catch((err) => {
+                    console.log("error", err)
+                    console.error('Failed to upload to IPFS');
+                    setFileUploading(false);
+                })
+            }).catch((err) => {
+                console.log("error", err)
+                console.error('Error uploading to IPFS:', err);
+                setFileUploading(false);
+            })
```

Comment:
Can you put errorManager to help on that and throw on sentry? (instead of console logs) and also make sure the functions inside are throwing error(so we can capture on errorManager).